apps/lib/apps.c

```diff
@@ -258,9 +258,9 @@ static char *app_get_pass(const char *arg, int keepbio)
     int i;
 
     /* PASS_SOURCE_SIZE_MAX = max number of chars before ':' in below strings */
-    if (strncmp(arg, "pass:", 5) == 0)
-        return OPENSSL_strdup(arg + 5);
-    if (strncmp(arg, "env:", 4) == 0) {
+    if (HAS_PREFIX(arg, "pass:"))
+        return OPENSSL_strdup(arg + sizeof("pass:") - 1);
+    if (HAS_PREFIX(arg, "env:")) {
         tmp = getenv(arg + 4);
```

Comment:
Maybe change 4 to `sizeof("env:") - 1`?

Reply:
An alternative would be to define the macro as:
```c
# define HAS_PREFIX(str, pre) ((strncmp(str, pre "", sizeof(pre) - 1) == 0) ? str + sizeof(pre) - 1 : NULL)
```
which allows:
```c
if ((tmp = HAS_PREFIX(...)) != NULL)
    tmp = getenv(tmp);
```
although this abuses our coding style unless the != NULL is made mandatory in cases without assignment.  A second macro for this case perhaps?

Reply:
If `HAS_PREFIX()` is made to return a pointer, then the result must *always* be explicitly checked against NULL.  I don't see that as a problem.

Reply:
I like the idea of supporting also the rather typical use case to advance the string pointer past the found prefix.
Adding a pointer result to `HAS_PREFIX` would complicate many uses due to the typically needed `!= NULL` check.
Moreover, this would stretch its semantics, so a more complicated name should be used then.
So IMO a second macro is better; I'll add
```
/* As before, and if check succeeds, advance the str ptr past the prefix */
# define CHECK_AND_SKIP_PREFIX(str, pre) \
    (HAS_PREFIX(str, pre) ? ((str) += sizeof(pre) - 1, 1) : 0)
```